File: Difficulty: Easy/Jumping Caterpillars/jumping-caterpillars.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    int uneatenLeaves(vector<int>& arr, int n) {
        vector<int> ans(n+1,0);
        for(int i=0;i<arr.size();i++)
        {
            int val=arr[i];
            while(val<=n)
            {
                if(ans[val]!=1)
                    ans[val]=1;
                val+=arr[i];
            }
        }
        int count=0;
        for(int i=1;i<=n;i++)
        {
            if(ans[i] == 0)
            {
                count++;
            }
        }        
        return count;
    }
};


//{ Driver Code Starts.

int main() {
    string ts;
    getline(cin, ts);
    int t = stoi(ts);
    while (t--) {
        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }
        string ks;
        getline(cin, ks);
        int k = stoi(ks);
        Solution obj;
        int res = obj.uneatenLeaves(arr, k);
        cout << res << endl;
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Easy/Jumping Caterpillars/jumping-caterpillars.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.lang.*;
import java.util.*;


// } Driver Code Ends

import java.util.*;

class Solution {
    public int uneatenLeaves(int[] arr, int n) {
        HashSet<Integer>set = new HashSet<>();
        for(int i=1; i<=n; i++){
            set.add(i);
        }
        int count = n;
        for(int i=0; i<arr.length; i++){
            int num = arr[i];
            while(num<=n){
                if(set.contains(num)){
                    set.remove(num);
                    count--;
                    
                }
                num+=arr[i];
                
            }
        }
        return count;
    }
}



//{ Driver Code Starts.

class GFG {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        while (t-- > 0) {
            String line = br.readLine();
            String[] tokens = line.split(" ");

            // Create an ArrayList to store the integers
            ArrayList<Integer> array = new ArrayList<>();

            // Parse the tokens into integers and add to the array
            for (String token : tokens) {
                array.add(Integer.parseInt(token));
            }
            int k = Integer.parseInt(br.readLine());

            int[] arr = new int[array.size()];
            int idx = 0;
            for (int i : array) arr[idx++] = i;
            Solution obj = new Solution();
            int res = obj.uneatenLeaves(arr, k);

            System.out.println(res);

            System.out.println("~");
        }
    }
}

// } Driver Code Ends